waspc/cli/src/Wasp/Cli/Command/Utils/File.hs

```diff
@@ -0,0 +1,26 @@
+module Wasp.Cli.Command.Utils.File
+  ( deleteFileIfExists,
+    deleteDirectoryIfExists,
+    getUserCacheDirPath,
+    UserCacheDir,
+  )
+where
+
+import Control.Monad (when)
+import qualified StrongPath as SP
+import qualified System.Directory as SD
+
+deleteDirectoryIfExists :: FilePath -> IO ()
+deleteDirectoryIfExists dirPath = do
+  doesDirExist <- SD.doesDirectoryExist dirPath
+  when doesDirExist $ SD.removeDirectoryRecursive dirPath
+
+deleteFileIfExists :: FilePath -> IO ()
+deleteFileIfExists dirPath = do
+  doesDirExist <- SD.doesFileExist dirPath
+  when doesDirExist $ SD.removeFile dirPath
+
+data UserCacheDir
+
+getUserCacheDirPath :: IO (SP.Path' SP.Abs (SP.Dir UserCacheDir))
+getUserCacheDirPath = SD.getXdgDirectory SD.XdgCache "" >>= SP.parseAbsDir
```

Comment:
Good idea getting these out! We can make them even better by swapping `FilePath` with Strong Path types. 

Context: Martin's comment in https://github.com/wasp-lang/wasp/pull/953#discussion_r1081063787 sparked a discussion in the office. We concluded that Strong Path gets clunky when we have to integrate it with regular file system functions and devised a plan to solve it for good: #972. This is a separate issue, but helps explain why it makes sense to change the signatures of these functions.

Reply:
> Good idea getting these out! We can make them even better by swapping FilePath with Strong Path types.

We are using StrongPath types here already, I'm not sure what else we can do to make it better here? :)

Reply:
Swap out `FilePath` with SP Paths:
```haskell
deleteDirectoryIfExists :: (Path a) b (Dir c) -> IO ()
deleteFileIfExists :: (Path a) b (File c) -> IO ()
```
You can even set the signatures to something less polymorphic if you think it makes sense. This makes sure that we can only pass directories to `deleteDirectoryIfExists` and only files to `deleteFileIfExists`.

Also, you have a typo in `abswWspBinFile`.

Reply:
Sorry, I didn't consider the whole file! Now that makes sense, and I've made the needed adjustment. I've went with your suggestion and did the most general typing.

Reply:
@sodic your StrongPath-fu is getting impressive, but still not 100% there :D :
It would be better to limit the "standard" of these paths to `System`, since if you are doing operations on the file system, you want the file paths standard to match the file system standard (I don't want to use Posix path to delete stuff on Win -> it won't work correctly).

So ideally signatures would be:
```hs
deleteDirectoryIfExists :: Path System r (Dir d) -> IO ()
deleteFileIfExists :: Path System r (File f) -> IO ()
```

These can be then shortened a bit with alias `Path'`:
```hs
deleteDirectoryIfExists :: Path' r (Dir d) -> IO ()
deleteFileIfExists :: Path' r (File f) -> IO ()
```